### PR TITLE
Feature new calendar

### PR DESCRIPTION
**Pull request by NguyenCQChi**
I added new calendar which can generate months and years

### CODE REVIEW

**Review by mbabey — APPROVED** [review]

File: calendar.html [hunk 1]
```diff
@@ -8,10 +8,113 @@
...
+            border: 1px solid #000;
+            box-shadow: 0px 0px 3px rgba(0, 0, 0, 0.2);
+        }
+        .version {
+            font-size: 12px;
+            width: 800px;
+            margin: 0 auto;
+            text-align: right;
+        }
+        </style>
```
Comment (nok-ko) [inline calendar.html]: Inline style! I'd rather have that in a stylesheet.

File: calendar.html [hunk 2]
```diff
@@ -97,6 +211,151 @@
             </div>
         </div>
     </footer>
+
+    <!-- Create the calendar -->
+<script type="text/javascript">
+    // Get elements
+    var elements = {
```
Comment (nok-ko) [inline calendar.html]: Avoid `var`, use `let`.

Comment (nok-ko) [reply calendar.html]:
```suggestion
    let elements = {
```

File: calendar.html [hunk 3]
```diff
@@ -97,6 +211,151 @@
...
+        // Calendar element
+        calendar : document.getElementById("events-calendar"),
+        // Input element
+        events : document.getElementById("events")
+    }
+
+    // Create the calendar
+    elements.calendar.className = "black-theme";
+    var calendar = jsCalendar.new(elements.calendar);
```
Comment (nok-ko) [inline calendar.html]: `let`

Comment (nok-ko) [reply calendar.html]:
```suggestion
    let calendar = jsCalendar.new(elements.calendar);
```

File: calendar.html [hunk 4]
```diff
@@ -97,6 +211,151 @@
...
+    elements.list = document.createElement("div");
+    elements.list.className = "list";
+    elements.events.appendChild(elements.list);
+    elements.actions = document.createElement("div");
+    elements.actions.className = "action";
+    elements.events.appendChild(elements.actions);
+    elements.addButton = document.createElement("input");
+    elements.addButton.type = "button";
+    elements.addButton.value = "Add";
+    elements.actions.appendChild(elements.addButton);
```
Comment (nok-ko) [inline calendar.html]: I don't like this, it makes it hard to edit the HTML. Can't you just include these elements in the HTML, but set the `hidden` property on them?

File: calendar.html [hunk 5]
```diff
@@ -97,6 +211,151 @@
...
+
+    var events = {};
+    var date_format = "DD/MM/YYYY";
+    var current = null;
+
+    var showEvents = function(date){
```
Comment (nok-ko) [inline calendar.html]: Why not `function showEvents(date) {…}`?

File: calendar.html [hunk 6]
```diff
@@ -97,6 +211,151 @@
...
+        // Date string
+        var id = jsCalendar.tools.dateToString(date, date_format, "en");
```
Comment (nok-ko) [inline calendar.html]: Why is this variable called `id`?

File: calendar.html [hunk 7]
```diff
@@ -97,6 +211,151 @@
...
+        // Set date
+        current = new Date(date.getTime());
+        // Set title
+        elements.title.textContent = id;
+        // Clear old events
+        elements.list.innerHTML = "";
+        // Add events on list
+        if (events.hasOwnProperty(id) && events[id].length) {
```
Comment (nok-ko) [inline calendar.html]: What does `events.hasOwnProperty(id)` do?

File: calendar.html [hunk 8]
```diff
@@ -97,6 +211,151 @@
...
+            // Number of events
+            elements.subtitle.textContent = events[id].length + " " + ((events[id].length > 1) ? "events" : "event");
+
+            var div;
+            var close;
```
Comment (nok-ko) [inline calendar.html]: No reason to have these declarations outside the loop.

File: calendar.html [hunk 9]
```diff
@@ -97,6 +211,151 @@
...
+                        removeEvent(date, index);
+                    }
+                })(date, i), false);
+            }
+        } else {
+            elements.subtitle.textContent = "No events";
+        }
+    };
+
+    var removeEvent = function (date, index) {
```
Comment (nok-ko) [inline calendar.html]: Again, why the `var someFunction = function (x,y) {…}` construct?

File: calendar.html [hunk 10]
```diff
@@ -97,6 +211,151 @@
...
+        }
+
+        // Date string
+        var id = jsCalendar.tools.dateToString(current, date_format, "en");
+
+        // If no events, create list
+        if (!events.hasOwnProperty(id)) {
+            // Create list
+            events[id] = [];
+        }
```
Comment (nok-ko) [inline calendar.html]: This smells like unnecessary code repetition.

Comment (nok-ko) [reply calendar.html]: I may be wrong, though.

File: styles/calendar_styles.css [hunk 11]
```diff
@@ -52,4 +52,485 @@ header {
...
+    }
+    .jsCalendar.white-theme tbody td.jsCalendar-next:hover,.jsCalendar.white-theme tbody td.jsCalendar-previous:hover {
+        color:#000
+    }
+    .jsCalendar.white-theme ::-moz-selection {
+        background:#AFAFAF
+    }
+    .jsCalendar.white-theme ::selection {
+        background:#AFAFAF
+    }
```
Comment (nok-ko) [inline styles/calendar_styles.css]: If this is all from the jsCalendar library, please move it to a separate file, or make this clear in a comment.

**Review by nok-ko — CHANGES_REQUESTED** [review]

**Pull request closed without merging**